src/harvester/wasser_de.rs

```diff
@@ -0,0 +1,83 @@
+use anyhow::Result;
+use cap_std::fs::Dir;
+use reqwest::Client;
+use serde::{Deserialize, Serialize};
+
+use crate::{dataset::Dataset, harvester::Source};
+
+pub async fn harvest(dir: &Dir, client: &Client, source: &Source) -> Result<()> {
+    let url = source
+        .url
+        .join("rest/BaseController/FilterElements/V_REP_BASE_VALID")?;
+
+    let response = client
+        .post(url)
+        .json(&Request { filter: Filter {} })
+        .send()
+        .await?
+        .error_for_status()?
+        .json::<Response>()
+        .await?;
+
+    tracing::info!("Retrieved {} documents", response.results.len().to_string());
```

Comment:
There is no need to call `.to_string()` as `tracing::info!` will take care of the formatting internally. This is all based on the standard library's [`fmt` module](https://doc.rust-lang.org/stable/std/fmt/index.html).